DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/EditDelegateControllerTests.cs

```diff
@@ -113,17 +153,70 @@ public void Index_post_returns_view_with_model_error_with_invalid_prn()
                     result,
                     "Invalid professional registration number format - Only alphanumeric characters (a-z, A-Z and 0-9) and hyphens (-) allowed"
                 );
+                A.CallTo(() => userService.GetDelegateById(A<int>._)).MustNotHaveHappened();
             }
         }
 
         [Test]
-        public void Index_post_calls_userService_and_redirects_with_no_validation_errors()
+        public void Index_post_calls_userServices_and_redirects_with_no_validation_errors()
         {
             // Given
+            const string centreSpecificEmail = "centre@email.com";
+            var delegateUser = UserTestHelper.GetDefaultDelegate(
+                DelegateId,
+                userCentreDetailsId: 1,
+                centreSpecificEmail: centreSpecificEmail,
+                centreSpecificEmailVerified: false
+            );
+            var formData = new EditDelegateFormData
+            {
+                JobGroupId = 1,
+                HasProfessionalRegistrationNumber = false,
+                CentreSpecificEmail = centreSpecificEmail,
+            };
+
+            A.CallTo(() => userService.GetDelegateById(DelegateId)).Returns(delegateUser);
+            A.CallTo(() => userService.NewEmailAddressIsValid(centreSpecificEmail, delegateUser.UserAccount.Id))
+                .Returns(true);
+
+            // When
+            var result = controller.Index(formData, DelegateId);
+
+            // Then
+            using (new AssertionScope())
+            {
+                A.CallTo(() => userService.GetDelegateById(DelegateId)).MustHaveHappenedOnceExactly();
+                A.CallTo(() => userService.NewEmailAddressIsValid(centreSpecificEmail, delegateUser.UserAccount.Id))
+                    .MustHaveHappenedOnceExactly();
+                A.CallTo(
+                    () => userService.UpdateUserDetailsAndCentreSpecificDetails(
+                        A<EditAccountDetailsData>._,
+                        A<DelegateDetailsData>._,
+                        centreSpecificEmail,
+                        delegateUser.DelegateAccount.CentreId
+                    )
+                ).MustHaveHappened();
+                result.Should().BeRedirectToActionResult().WithControllerName("ViewDelegate").WithActionName("Index");
+            }
+        }
+
+        [Test]
+        public void Index_post_saves_centre_specific_email_as_null_if_same_as_primary_email()
+        {
+            // Given
+            const string primaryEmail = "primary@email";
+            var delegateUser = UserTestHelper.GetDefaultDelegate(

```

Comment:
`var delegateUser` is a `Delegate` rather than a `DelegateUser`, so should be renamed